Setting up macOS as an R data science rig in 2023

ivelasq opened the pull request:
@gvelasq , please find the PR for the blog post here.

ivelasq committed:
Start post

ivelasq committed:
Start post

ivelasq committed:
Add to post

ivelasq committed:
Add to post

gvelasq commented on blog/macos-rig/index.qmd:
```diff
@@ -0,0 +1,309 @@
+title: "Setting up macOS as an R data science rig in 2023"
+date: "2022-12-19"
+categories: ["tutorial"]
+description: "Let’s fall together into a pit of success (when configuring macOS)."
```
Suggest to end with an exclamation mark rather than a period.

gvelasq replied:
Wherever pit of success is mentioned, add a hyperlink to: https://blog.codinghorror.com/falling-into-the-pit-of-success

gvelasq commented on blog/macos-rig/index.qmd:
```diff
@@ -0,0 +1,309 @@
+
+Like a new year 🗓️, a new computer 🖥️ (or a fresh reinstall) can bring so many opportunities. A clear desktop! So much memory! And then you remember that you have to install and configure all of your programs again, and that initial excitement gives way to despair and hopelessness. 😱
+
+Okay, that’s a bit of an exaggeration, 🤭 but it can be tedious to reconfigure your setup the way that you like it. However, by using 'dotfiles,' or hidden configuration files (whose names start with a dot/period, which on macOS hides them from view in the Finder), you can store your user preferences on a hosted version control service like GitHub. By dropping dotfiles in the correct directory, yo ...
```
Suggest to remove single quotes around dotfiles.

gvelasq replied:
Suggest to add hyperlink to GitHub.

gvelasq commented on blog/macos-rig/index.qmd:
```diff
@@ -0,0 +1,309 @@
+
+Okay, that’s a bit of an exaggeration, 🤭 but it can be tedious to reconfigure your setup the way that you like it. However, by using 'dotfiles,' or hidden configuration files (whose names start with a dot/period, which on macOS hides them from view in the Finder), you can store your user preferences on a hosted version control service like GitHub. By dropping dotfiles in the correct directory, yo ...
+
+In this post, we’ll share some helpful dotfiles for all you RStudio users who are rocking a new macOS 🍎installation in 2023. This post is geared towards you if:
```
Suggest to add a space after the apple emoji.

gvelasq commented on blog/macos-rig/index.qmd:
```diff
@@ -0,0 +1,309 @@
+* You want a one-stop-shop for setting up your rig next time;
+* You have a well-oiled, well-defined workflow but are interested in seeing how others set up their machines, to perhaps learn something new!
+
+Since it is the year of [ChatGPT](https://openai.com/blog/chatgpt/), [we did ask its suggested setup](#chat-gpt). We thought we could do better (at least in 2022, maybe not in 2024)! 🤗
```
Suggest to change 'at least in 2022' to 'at least _for now_ in 2022'.

ivelasq replied:
Changed it to "at least for now, maybe not in 2024".

gvelasq commented on blog/macos-rig/index.qmd:
```diff
@@ -0,0 +1,309 @@
+
+3. Configure [global Git settings](https://github.com/gvelasq/git-reference#setup).
+
+Run the below (replacing the text in the quotation marks with your information) to adjust your Git settings.
```
Suggest to change 'Run the below' to 'Run the code below'.

gvelasq commented on blog/macos-rig/index.qmd:
```diff
@@ -0,0 +1,309 @@
+
+4. Install [zsh](https://zsh.sourceforge.io).
+
+Z shell, or zsh, is a useful tool for interactive use in the Terminal. Several years back, macOS switched from using the bash shell to using zsh as the default shell. Installing the latest version of zsh from Homebrew gives you the *latest* version — otherwise, zsh and Git are both stuck at the version bundled with the last time you have updated macOS — which makes them open to vulnerabilities an ...
```
Suggest to delete the first sentence: 'Z shell, or zsh, is a useful tool for interactive use in the Terminal.'

gvelasq replied:
Suggest change 'version bundled with' to 'versions bundled' (remove the 'with').

gvelasq replied:
Suggest to change 'have updated' to 'updated'.

gvelasq commented on blog/macos-rig/index.qmd:
```diff
@@ -0,0 +1,309 @@
+
+{{< tweet robbyrussell 1599596073282342914 >}}
+
+It comes with a ton of plugins, themes, and functions. For most users, having the great themes available from OMZ are enough to switch. But, the real value is [all the built-in OMZ plugins](https://github.com/ohmyzsh/ohmyzsh/wiki/Plugins) — which provide a rich set of aliases for Homebrew, Git, and many other command-line programs. For example, `gs` is `git status`, `bubo` is `brew update && brew ...
```
Suggest to change 'But,' to 'But for us,'.

gvelasq replied:
For all of the plugins mentioned below, instead of showing the full links to the plugins, suggest to hyperlink them to the word in the list. For example, hyperlink the brew plugin to the word 'brew' in the first bullet, the git plugin to the word 'git' in the second bullet, etc.

ivelasq committed:
Add authors, italics, and caption

gvelasq commented on blog/macos-rig/index.qmd:
```diff
@@ -0,0 +1,309 @@
+* The macos plugin for easier living through the Terminal: [https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/macos](https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/macos) 
+* There's even a plugin that reminds you what your aliases are! -- for maximum discoverability: [https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/aliases](https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/aliases)
+
+You can also hand-code your themes into your `.zshrc` dotfile instead of using the OMZ themes. The aliases not provided by OMZ can be added to the custom.zsh dotfile, such as the ones in the dotfiles linked below.
```
Suggest to add backticks around `custom.zsh`.

gvelasq commented on blog/macos-rig/index.qmd:
```diff
@@ -0,0 +1,309 @@
+
+![](image1.jpg){fig-align="center" fig-alt="Keep Calm and bubo bog bupc bc" fig-width="400"}
+
+7. Go back to the Terminal and install [rig](https://github.com/r-lib/rig#linux).
```
Right now the hyperlink points to the Linux installation section. Suggest to change the link to the macOS Homebrew section: https://github.com/r-lib/rig#macos-homebrew

gvelasq commented on blog/macos-rig/index.qmd:
```diff
@@ -0,0 +1,309 @@
+
+7. Go back to the Terminal and install [rig](https://github.com/r-lib/rig#linux).
+
+[rig](https://github.com/r-lib/rig#linux), the 'R Installation Manager,' is an amazing tool. It not only lets you download R, but install, configure, switch between, and remove _multiple_ R versions. It also allows you to manage _user-level package libraries_ (meaning you can delete a version of R without losing all your packages!). Mara Averick wrote a great summary on [Day 10 of Tidy Advent 202 ...
```
Suggest to change rig hyperlink to point to https://github.com/r-lib/rig

gvelasq commented on blog/macos-rig/index.qmd:
````diff
@@ -0,0 +1,309 @@
+brew install --cask rig
+```
+
+8. Install R.
````
Suggest to add link to R: https://www.r-project.org

gvelasq commented on blog/macos-rig/index.qmd:
````diff
@@ -0,0 +1,309 @@
+brew install --cask rstudio
+```
+
+10. <a name="abcd">Configure [`.Rprofile`](https://github.com/gvelasq/dotfiles/blob/main/R/.Rprofile).</a>
````
Why is there HTML around this line? `<a name="abcd"></a>`

ivelasq replied:
To create an anchor link. I don't know how to do it in Markdown.

gvelasq commented on blog/macos-rig/index.qmd:
````diff
@@ -0,0 +1,309 @@
+usethis::edit_r_profile()
+```
+
+[Here's an example of an `.Rprofile` script.](https://github.com/gvelasq/dotfiles/blob/main/R/.Rprofile) This script [1] sets {devtools} and {usethis} to load with each R session (to write `install_github()` without a `library(devtools)`), [2] sets [RSPM](https://packagemanager.rstudio.com/client/#/) as the default repo for Linux (not relevant for this macOS guide), [3] sets a bunch of {usethis}  ...
````
Should RSPM now be PPPM? In which case perhaps the full acronym should be spelled out with PPPM in parens.

gvelasq commented on blog/macos-rig/index.qmd:
```diff
@@ -0,0 +1,309 @@
+
+![](gif1.gif){fig-align="center"}
+
+For whenever you want to remember De Morgan's laws 😉
```
Suggest to add a period after 'laws'.

gvelasq commented on blog/macos-rig/index.qmd:
````diff
@@ -0,0 +1,309 @@
+```r
+install.packages("gitcreds")
+```
+ Manage your GitHub tokens or passwords:
````
This is actually a two-step procedure so this section needs to be reworded. First, `usethis::create_github_token()` will (per the help file): 'open a browser window to the GitHub form to generate a PAT, with suggested scopes pre-selected' -- the first part is key because it's helpful to navigate there in one command, the last part is key because it helps you make the right choice of scopes every time. Then, the command will suggest to use `gitcreds::gitcreds_set()` to save the PAT locally using the macOS Git credential manager (after copying it from GitHub).

gvelasq commented on blog/macos-rig/index.qmd:
```diff
@@ -0,0 +1,309 @@
+
+Quarto is a (kinda new) open-source scientific and technical publishing system built on Pandoc. It allows you to create dynamic content with R and Python, like documents, websites (like this blog!), and [presentations](https://ivelasq.rbind.io/talk/intro-to-quarto/).
+
+Quarto comes bundled with RStudio v. 2022.07 and later, so the installation in [10](#abcd) should have this covered. But, if you’d like to install Quarto yourself, run this in your Terminal:
```
Suggest to change '10 should' to 'step 10 above should'.

gvelasq replied:
I see now the HTML was for abcd to link here, suggest that in step 10 it's not around the entire Configure... but just the .Rprofile?

ivelasq replied:
Should it be step 9?

gvelasq commented on blog/macos-rig/index.qmd:
```diff
@@ -0,0 +1,309 @@
+
+## Final Thoughts
+
+We hope that this post is helpful in setting up an R data science rig on your Mac! We know there were a lot of steps. In summary:
```
Suggest to change 'on your Mac' to 'on macOS'.

gvelasq commented on blog/macos-rig/index.qmd:
```diff
@@ -0,0 +1,309 @@
+We hope that this post is helpful in setting up an R data science rig on your Mac! We know there were a lot of steps. In summary:
+
+* Dotfiles are hidden configuration files where you can store your user preferences and use them across machines and installations.
+* The configuration steps above ensure that you're using the best tools for the Mac terminal: Homebrew (for package management) and zsh (for execution).
```
Suggest to change 'Mac' to 'macOS'.

gvelasq commented on blog/macos-rig/index.qmd:
```diff
@@ -0,0 +1,310 @@
+
+Since it is the year of [ChatGPT](https://openai.com/blog/chatgpt/), [we did ask its suggested setup](#chat-gpt). We thought we could do better (at least in 2022, maybe not in 2024)! 🤗
+
+Let’s fall into a pit of success <sup>TM</sup> (when configuring macOS)!
```
For 'pit of success', add hyperlink to: https://blog.codinghorror.com/falling-into-the-pit-of-success

ivelasq committed:
Update post

ivelasq committed:
Update date

ivelasq committed:
Rerender blog

Merge remote-tracking branch 'origin/main' into mac-confi

# Conflicts:
#	_site/index.html
#	_site/index.xml
#	_site/sitemap.xml

ivelasq committed:
Make small edits

gvelasq commented:
LGTM!

ivelasq committed:
Render post
